egs/hub4_english/s5/local/prepare_dict.sh

```diff
@@ -0,0 +1,213 @@
+# its output in data/local/.
+
+# The parts of the output of this that will be needed are
+# [in data/local/dict/ ]
+# lexicon.txt
+# extra_questions.txt
+# nonsilence_phones.txt
+# optional_silence.txt
+# silence_phones.txt
+
+[ -f ./path.sh ] && . ./path.sh
+. ./cmd.sh
+
+set -e
+set -o pipefail
+set -u
+
+# run this from ../
+dict_suffix=
+stage=-1
+
+echo "$0 $@"  # Print the command line for logging
+. utils/parse_options.sh || exit 1;
+
+if [ $# -ne 1 ]; then
+  echo "Usage: $0 <wordlist>"
+  echo "e.g. : $0 data/local/local_lm/data/work/wordlist"
+  exit 1
+fi
+
+wordlist=$1
+
+dir=data/local/dict${dict_suffix}
+mkdir -p $dir
+
+if [ ! -d $dir/cmudict ]; then
+  # (1) Get the CMU dictionary
+  svn co  https://svn.code.sf.net/p/cmusphinx/code/trunk/cmudict \
+    $dir/cmudict || exit 1;
+fi
+
+# can add -r 10966 for strict compatibility.
+
+
+#(2) Dictionary preparation:
+
+
+if [ $stage -le 0 ]; then
+  # Make phones symbol-table (adding in silence and verbal and non-verbal noises at this point).
+  # We are adding suffixes _B, _E, _S for beginning, ending, and singleton phones.
+
+  # silence phones, one per line.
+  (echo SIL; echo SPN; echo NSN; echo UNK;) > $dir/silence_phones.txt
+  echo SIL > $dir/optional_silence.txt
+
+  # nonsilence phones; on each line is a list of phones that correspond
+  # really to the same base phone.
+  cat $dir/cmudict/cmudict.0.7a.symbols | perl -ane 's:\r::; print;' | \
+    perl -e 'while(<>){
+  chop; m:^([^\d]+)(\d*)$: || die "Bad phone $_";
+  $phones_of{$1} .= "$_ "; }
+  foreach $list (values %phones_of) {print $list . "\n"; } ' \
+    > $dir/nonsilence_phones.txt || exit 1;
+
+  # A few extra questions that will be added to those obtained by automatically clustering
+  # the "real" phones.  These ask about stress; there's also one for silence.
+  cat $dir/silence_phones.txt| awk '{printf("%s ", $1);} END{printf "\n";}' > $dir/extra_questions.txt || exit 1;
+  cat $dir/nonsilence_phones.txt | perl -e 'while(<>){ foreach $p (split(" ", $_)) {
+  $p =~ m:^([^\d]+)(\d*)$: || die "Bad phone $_"; $q{$2} .= "$p "; } } foreach $l (values %q) {print "$l\n";}' \
+    >> $dir/extra_questions.txt || exit 1;
+
+  grep -v ';;;' $dir/cmudict/cmudict.0.7a | \
+    perl -ane 'if(!m:^;;;:){ s:(\S+)\(\d+\) :$1 :; print; }' \
+    > $dir/dict.cmu || exit 1;
+
+  # Add to cmudict the silences, noises etc.
+
+  (echo '!SIL SIL'; echo '<SPOKEN_NOISE> SPN'; echo '<unk> UNK'; echo '<NOISE> NSN'; ) | \
+    cat - $dir/dict.cmu > $dir/lexicon2_raw.txt
+  awk '{print $1}' $dir/lexicon2_raw.txt > $dir/orig_wordlist
+
+  cat <<EOF >$dir/silence_phones.txt
+SIL
+SPN
+NSN
+UNK
+EOF
+
+fi
+
+
+if [ $stage -le 2 ]; then
+  if [ ! -f exp/g2p/.done ]; then
+    steps/dict/train_g2p.sh --cmd "$train_cmd" \
+      --silence-phones $dir/silence_phones.txt \
+      $dir/dict.cmu exp/g2p
+    touch exp/g2p/.done
+  fi
+fi
+
+export PATH=$PATH:`pwd`/local/dict
+
+if [ $stage -le 3 ]; then
+  cat $wordlist | python -c '
```

Comment:
I think you could replace this inline python script with:
`utils/filter_scp.pl --exclude $wordlist <$orig_wordlist > $dir/oovlist`